pkg/daemon/controller/controller.go

```diff
@@ -377,6 +360,44 @@ func (sc *ServiceController) resolvePowershellVariables(svc servicescm.Service)
 	return make(map[string]string), nil
 }
 
+// NewDirectClient creates and returns an authenticated client that reads directly from the API server.
+// It also returns the config and shceme used to created the client.
+func NewDirectClient(apiServerURL, saCA, saToken string) (client.Client, *rest.Config, *runtime.Scheme, error) {
+	cfg, err := config.FromServiceAccount(apiServerURL, saCA, saToken)
+	if err != nil {
+		return nil, nil, nil, errors.Wrap(err, "error using service account to build config: %s")
+	}
+
+	clientScheme := runtime.NewScheme()
+	err = clientgoscheme.AddToScheme(clientScheme)
+	if err = clientgoscheme.AddToScheme(clientScheme); err != nil {
+		return nil, nil, nil, err
+	}
+
+	directClient, err := client.New(cfg, client.Options{Scheme: clientScheme})
+	if err != nil {
+		return nil, nil, nil, err
+	}
+	return directClient, cfg, clientScheme, nil
+}
+
+// GetUnderlyingNode uses the given client to find the name of the node associated with the VM this is running on
+func GetUnderlyingNode(c client.Client) (*core.Node, error) {
```

Comment:
Consider renaming this `GetInstanceNode`/ `GetAssociatedNode`, or something similar. I'd argue `Overlying` would be more correct than `Underlying`, as the node is an abstraction layer on top of the instance :wink: .